subsys/bluetooth/host/att_internal.h

```diff
@@ -263,14 +263,19 @@ typedef void (*bt_att_func_t)(struct bt_conn *conn, uint8_t err,
 			      const void *pdu, uint16_t length,
 			      void *user_data);
 
+typedef void (*bt_att_write_buf_t)(struct net_buf *buf, size_t len,
+				   void *user_data);
+
 /* ATT request context */
 struct bt_att_req {
 	sys_snode_t node;
 	bt_att_func_t func;
-	struct net_buf_simple_state state;
 	struct net_buf *buf;
 #if defined(CONFIG_BT_SMP)
-	bool retrying;
+	bt_att_write_buf_t write_buf;
```

Comment:
I wonder if it wouldn't be a better idea to push the resend logic to gatt.c then, since now we would need to callbacks back to it to encode the request again it might be better to have everything there and perhaps have something like bt_att_resend to prepend the request on the queue, anyway we can make this change later as well if we find it necessary.